exercises/concept/card-tricks/card_tricks_test.go

```diff
@@ -133,121 +132,141 @@ func TestSetItem(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := SetItem(tt.args.slice, tt.args.index, tt.args.value); !slicesEqual(got, tt.want) {
+			got := SetItem(tt.args.slice, tt.args.index, tt.args.value)
+			if !slicesEqual(got, tt.want) {
 				t.Errorf("SetItem(slice:%v, index:%v, value:%v) = %v, want %v",
 					tt.args.slice, tt.args.index, tt.args.value, got, tt.want)
 			}
+			if len(tt.args.slice) == len(got) {
+				for i := range got {
+					got[i] = -1
+				}
+				if !slicesEqual(got, tt.args.slice) {
+					t.Errorf("SetItem(slice:%v, index:%v) does not return the modified input slice)", tt.args.slice,
+						tt.args.value)
+				}
```

Comment:
since we already established the values are equal, I would recommend something like this to check the reference is still the same: https://go.dev/play/p/oHmY2lUpO5L

Reply:
Nice! Done.